pkg/controller/datadogagent/utils.go

```diff
@@ -159,6 +159,17 @@ func getAgentContainer(dda *datadoghqv1alpha1.DatadogAgent) (*corev1.Container,
 	if err != nil {
 		return nil, err
 	}
+
+	udpPort := corev1.ContainerPort{
+		ContainerPort: datadoghqv1alpha1.DefaultDogstatsdPort,
```

Comment:
It will also required to add the `DD_DOGSTATSD_PORT` env-var to the "agent" container to override the default value if never the value provided it not "8125"

Reply:
Do we necessarily want to have `DD_DOGSTATSD_PORT` match the `hostPort` ? 
I think we should definitely support this, but this PR was intended more towards setting `hostPort` (the port that is bound on the host). 

I can see cases where you'd want to have `hostPort=18125` for example but not need to change the `DD_DOGSTATSD_PORT`, since it's "abstracted-away" in the container.

Reply:
The problem comes when the user also set: `hostNetwork`

Reply:
@clamoriniere Implemented this.
I'm now thinking we could only add that variable when both `hostPort` and `hostNetwork` are set. What do you think?

Reply:
Yes,  this is is a good idea.


```golang
	udpPort := corev1.ContainerPort{
		ContainerPort: datadoghqv1alpha1.DefaultDogstatsdPort,
                 Name:          "dogstatsdport",
		Protocol:      corev1.ProtocolUDP,
	}
	if agentSpec.Config.HostPort != nil && datadoghqv1alpha1.BoolValue(agentSpec.Config.HostNetwork) {
	    udpPort.ContainerPort = *agentSpec.Config.HostPort
	    // TODO add also the envvar
        }

Reply:
@clamoriniere Hey! Could you review my changes? I've implemented something similar to your suggestion.

Reply:
Hi @thehunt33r
sorry for the delay. 
@ahmed-mez could you make a final review?